R/basic.R

```diff
@@ -22,6 +25,15 @@ amqp_publish <- function(conn, body, exchange = "", routing_key = "",
   if (!inherits(conn, "amqp_connection")) {
     stop("`conn` is not an amqp_connection object")
   }
+  if (inherits(body, "character")) {
```

Comment:
`is.character()` is a better choice here, since (1) `charToRaw` uses the storage mode, not the S3 class, and (2) it's much faster.